ds3-autogen-java/src/main/resources/tmpls/models/bulk_object_template.ftl

```diff
@@ -37,17 +37,23 @@ ${javaHelper.getModelVariable(elmt)}
 
         final ${name} bulkObject = (${name}) obj;
 
-        return (((this.getId() == null) && (bulkObject.getId() == null))
-                || ((this.getId() != null) && (bulkObject.getId() != null) && this.getId().equals(bulkObject.getId())))
-                && this.getInCache().equals(bulkObject.getInCache())
+        return nullableEquals(this.getId(), bulkObject.getId())
+                && nullableEquals(this.getInCache(), bulkObject.getInCache())
                 && this.getLatest() == bulkObject.getLatest()
                 && this.getLength() == bulkObject.getLength()
-                && this.getName().equals(bulkObject.getName())
+                && nullableEquals(this.getName(), bulkObject.getName())
                 && this.getOffset() == bulkObject.getOffset()
                 && this.getPhysicalPlacement() == bulkObject.getPhysicalPlacement()
                 && this.getVersion() == bulkObject.getVersion();
     }
 
+    /**
+     * Tests if two objects are equal, and handles the case if either or both objects are null
+     */
+    protected static boolean nullableEquals(final Object obj1, final Object obj2) {
```

Comment:
None that I can currently think of.  I created a separate function to clean up the comparison code because the parentheses were getting rather nested and ugly. I can move it to guard if you want.